internal/services/catalog_service.go

```diff
@@ -32,6 +32,190 @@ type CatalogService struct {
+
+	for _, servicePlan := range servicePlans.Items {
+		spec := servicePlan.Object["spec"].(map[string]interface{})
+
+		externalName := spec["externalName"].(string)
+		description := spec["description"].(string)
+		isAFreePlan := spec["free"].(bool)
+
+		result = append(result, ServicePlan{
+			Name:        externalName,
+			Description: description,
+			Free:        isAFreePlan,
+		})
+	}
+
+	return result, nil
+
+	// spec:
+	//   clusterServiceBrokerName: minibroker
+	//   clusterServiceClassRef:
+	//     name: mongodb
+	//   description: NoSQL document-oriented database that stores JSON-like documents with
+	//     dynamic schemas, simplifying the integration of data in content-driven applications.
+	//   externalID: mongodb-4-0-12
+	//   externalName: 4-0-12
+	//   free: true
+}
+
+// ListClasses returns a ServiceClassList of all available catalog service classes
+func ListClasses(kubeClient *kubernetes.Cluster) (ServiceClassList, error) {
+
+	serviceClassGVR := schema.GroupVersionResource{
+		Group:    "servicecatalog.k8s.io",
+		Version:  "v1beta1",
+		Resource: "clusterserviceclasses",
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(kubeClient.RestConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	serviceClasses, err := dynamicClient.Resource(serviceClassGVR).
+		List(context.Background(),
+			metav1.ListOptions{})
+
+	if err != nil {
+		return nil, err
+	}
+
+	result := ServiceClassList{}
+
+	for _, serviceClass := range serviceClasses.Items {
+		spec := serviceClass.Object["spec"].(map[string]interface{})
+
+		externalName := spec["externalName"].(string)
+		description := spec["description"].(string)
+		clusterServiceBrokerName := spec["clusterServiceBrokerName"].(string)
+
+		metadata := serviceClass.Object["metadata"].(map[string]interface{})
+
+		labels := metadata["labels"].(map[string]interface{})
+		hash := labels["servicecatalog.k8s.io/spec.externalID"].(string)
+
+		result = append(result, ServiceClass{
+			Name:        externalName,
+			Broker:      clusterServiceBrokerName,
+			Description: description,
+			Hash:        hash,
+			kubeClient:  kubeClient,
+		})
+	}
+
+	return result, nil
+
+	//	metadata:
+	//	  creationTimestamp: "2021-03-15T13:06:46Z"
+	//	  generation: 1
+	//	  labels:
+	//	    servicecatalog.k8s.io/spec.clusterServiceBrokerName: d42e833047334b72abd9b8d804264d4dc8077517a5ef767ee76ba44d
+	// *1	    servicecatalog.k8s.io/spec.externalID: 431ba23818d0e82e05612ddd029dfb1506d2f255d96205606e1ee464
+	//	    servicecatalog.k8s.io/spec.externalName: 431ba23818d0e82e05612ddd029dfb1506d2f255d96205606e1ee464
+	//	spec:
+	//	  bindable: true
+	//	  bindingRetrievable: false
+	// **	  clusterServiceBrokerName: minibroker
+	// **	  description: Helm Chart for redis
+	//	  externalID: redis
+	// **	  externalName: redis
+	//	  planUpdatable: false
+	//
+	// Ad *1: This hash is the value of the
+	// 	`servicecatalog.k8s.io/spec.clusterServiceClassRef.name`
+	// label of the assocoiated plans.
```

Comment:
same (comments)

Reply:
I had left them in as example of the structure the code is digging through to get the information.

Reply:
Removed with commit 88d5750c76298888fa30a556e25ae969088be806